jscomp/core/js_exp_make.ml

```diff
@@ -316,6 +316,13 @@ let small_int i : t =
   | 248 -> obj_int_tag_literal
   | i -> int (Int32.of_int i)
 
+let as_value = function
+  | Lambda.AsString s -> str s ~delim:DStarJ
+  | AsInt i -> small_int i
+  | AsNull -> nil
+  | AsUndefined -> undefined
+  | AsUnboxed -> assert false (* Should not emit tags for unboxed *)
```

Comment:
TODO: put restriction on the variant definitions allowed, to make sure this never happens.